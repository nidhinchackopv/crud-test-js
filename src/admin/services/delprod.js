export async function delData(id) {
    const data = await fetch(`http://localhost:3000/users/${id}`,{
                 method: 'DELETE'
            });
        return data;
}