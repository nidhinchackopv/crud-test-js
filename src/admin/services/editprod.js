

// export async function editProd(props) {
//     const data = await fetch(`http://localhost:3000/users/${props.match.params.id}`, {
//             method: 'PUT',
//             headers: {'Content-Type':'application/json'},
//             body: JSON.stringify({
//                 title: props.title,
//                 body: props.body
//             })
//         })
//  return data;
// }

// export async function editProdOne(id) {
//         const response = await fetch(`http://localhost:3000/users/${props.match.params.id}`);

//         const data = await response.json();
//         return data;
// }