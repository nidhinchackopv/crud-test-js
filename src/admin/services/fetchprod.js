export async function fetchData() {

    const response = await fetch("http://localhost:3000/users")
    const data = await response.json();
    return data;
}