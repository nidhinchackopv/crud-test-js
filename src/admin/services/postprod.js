export async function getData(title,body) {
    const data = await fetch("http://localhost:3000/users", {
            method: 'POST',
            headers: {'Content-Type':'application/json'},
            body: JSON.stringify({
                title,
                body
            })
        })
 return data;
}