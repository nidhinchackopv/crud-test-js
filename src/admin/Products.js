import { act } from '@testing-library/react';
import React, { useEffect , useState} from 'react'
import { Link, BrowserRouter } from 'react-router-dom'
import * as APICall from './services/fetchprod'
import * as APIDelete from './services/delprod'


const Products = (props) => {

    const [products, setProducts] = useState([]);

    useEffect( () =>{
        (
            async () => {
                const data = await APICall.fetchData()

               setProducts(data);
            }
        )();
    },[]);

    const del = async (id) => {

        if(window.confirm("Are you sure you want to delete it")){

            const data = await APIDelete.delData(id)

            setProducts(products.filter((p) => p.id !== id ));
            // window.location.reload()
        }

    }
    return (

        <div className="shadow-lg bg-purple-200 pt-10 pb-20">
        <a href="/products/create" data-test-id = 'pro' className="shadow-md border-solid rounded-xl border-4 border-green-200 bg-purple-500 p-2">Add Information</a>
            <h1 className=" pt-10 align-center w-100 h-20 font-sans font-medium antialiased text-2xl">All the information</h1>
            <div className="pt-5 w-full">
                <table data-testid='table' className=" ml-2 mr-2 pt-5 w-full border-separate border border-green-800">
                <thead>
                    <tr className="">
                    <th className="">#id</th>
                    <th className="">Title</th>
                    <th className="">Body</th>
                    </tr>
                </thead>
                <tbody data-testid="tbody">
                    
                    {
                       products.map((p) => {
                            return (
                                <tr key={p.id}>
                                <td>{p.id}</td>
                                <td>{p.title}</td>
                                <td>{p.body}</td>
                                <td>
                                    <div className='mb-5'>
                                        <a href={`/products/${p.id}/edit`} data-testid="edit"
                                        className="shadow-md border-solid rounded-xl m-3 border-4 border-gray-200 bg-purple-500 p-2">Edit</a>
                                    </div>
                                </td>
                                <td>
                                    <div className='mb-5'>
                                        <a key= "delete" href="#" data-testid = "delete" className="shadow-md border-solid rounded-xl border-4 border-gray-200 bg-purple-500 p-2"
                                        onClick={() => del(p.id)} >Delete</a>
                                    </div>
                                </td>
                                </tr>
                                
                            )
                       })
                    }
                    
                </tbody>
                </table>
            </div>
            </div>
    )
}

export default Products
