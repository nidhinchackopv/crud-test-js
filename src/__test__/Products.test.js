import React from 'react';
import { fireEvent, render, screen, act, cleanup, waitFor, getByText } from '@testing-library/react';
import * as ReactDom from 'react-dom'
import Products from '../admin/Products'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import * as APICall from '../admin/services/fetchprod'
import * as APIDelete from '../admin/services/delprod'


jest.mock('../admin/services/fetchprod');

it('fetching', async () => {
  APICall.fetchData.mockResolvedValueOnce({ok : true});
  render(<Products />);
  expect(APICall.fetchData).toHaveBeenCalledTimes(1)
})

jest.mock('../admin/services/delprod');

// it('fetching', async () => {
//   APIDelete.delData.mockResolvedValueOnce({ok : true});
//   render(<Products key = {"delete"}/>)
//   // expect(fireEvent.click(containerone.querySelector("[data-testid = 'delete']")))
//   const button = screen.findByTestId('delete')
//   fireEvent.click(button)
//   expect(APIDelete.delData).toHaveBeenCalledTimes(1)
// })


describe("component test", () => {
  let container
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    ReactDom.render(<Products />, container);
  })

  afterEach(() => {
    document.body.removeChild(container);
    container.remove(),
    cleanup
  })


  it("table delete and edit", () => {
    render(<Products />)
    expect(container.querySelector("[data-testid = 'table']"))
    expect(fireEvent.change(container.querySelector("[data-testid = 'tbody']")))
    expect(container.querySelector("[data-testid = 'delete']"))
    expect(container.querySelector("[data-testid = 'edit']"))
  })
  
})

test("table items", () => {
  render(<Products />)
  expect(screen.getByText("#id"))
  expect(screen.getByText("Title"))
  expect(screen.getByText("Body"))
})

it('can render the About page', () => {
  const tree = renderer.create(<Products />).toJSON();
  expect(tree).toMatchSnapshot();
})

  // test('Anchor needs to have proper text', () => {
  //   const element = shallow(<App />);
  //   expect(element.find('a').text()).toBe("Edit");
  // });

//   it("calls the `openPopup` function", () => {
//     const onMockFunction = jest.fn();
//     const comp = shallow(
//         <SparkTheme>
//             <Contact handleButtonClick={onMockFunction} />
//         </SparkTheme>
//     );
//     comp.find(Header).simulate("change");
//     expect(onMockFunction).toBeCalledTimes(1);
// });

// it('test', async () => {
//     let component;
//     const fakeResponse = 'example text';
//     const mockFetch = Promise.resolve({json: () => Promise.resolve(fakeResponse)});
//     const mockedFetch = jest.spyOn(window, 'table').mockImplementationOnce(() => mockFetch )
//     await waitFor( async () => {
//         component = render(<Products />);
//     })
//     const value = component.container.querySelector("[data-testid = 'tbody']")
//     console.log(value);
//     expect(mockedFetch).toHaveBeenCalledTimes(1);
//   })

// global.fetch = jest.fn(() => Promise.resolve({
//   json: () => Promise.resolve({
//     title:"how are you"
//   })
// }))
// describe("fetch", () => {
// test('fetching',async () => {
//   await act( async () => render(<Products />))
//   expect(screen.getByText("how are you"))
// })
// })