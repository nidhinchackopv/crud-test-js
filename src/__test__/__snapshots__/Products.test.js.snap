// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render the About page 1`] = `
<div
  className="shadow-lg bg-purple-200 pt-10 pb-20"
>
  <a
    className="shadow-md border-solid rounded-xl border-4 border-green-200 bg-purple-500 p-2"
    data-test-id="pro"
    href="/products/create"
  >
    Add Information
  </a>
  <h1
    className=" pt-10 align-center w-100 h-20 font-sans font-medium antialiased text-2xl"
  >
    All the information
  </h1>
  <div
    className="pt-5 w-full"
  >
    <table
      className=" ml-2 mr-2 pt-5 w-full border-separate border border-green-800"
      data-testid="table"
    >
      <thead>
        <tr
          className=""
        >
          <th
            className=""
          >
            #id
          </th>
          <th
            className=""
          >
            Title
          </th>
          <th
            className=""
          >
            Body
          </th>
        </tr>
      </thead>
      <tbody
        data-testid="tbody"
      />
    </table>
  </div>
</div>
`;
